ipamd/datastore/data_store.go

```diff
@@ -343,6 +387,29 @@ func (ds *DataStore) FreeENI() (string, error) {
 	return deletedENI, nil
 }
 
+// DeleteENI free a ENI.
+func (ds *DataStore) DeleteENI(eni string) error {
+	ds.lock.Lock()
+	defer ds.lock.Unlock()
+
+	_, ok := ds.eniIPPools[eni]
```

Comment:
Save this in a variable here, `eniIPPool, ok = ds.eniIPPools[eni]` and replace the three lookups below.

Reply:
done